Run tests periodically and send email on failure

smiclea opened the pull request:
The integration tests have been cherry-picked (and squashed) from the `edge` branch.

claudiubelu committed:
Adds FINOS Legend Bundle Integration tests

Adds the linting and integration test github workflows which will be triggered
when a PR is created, when it merges, and when a manual workflow is dispatched.

The integration tests expect the GITLAB_CLIENT_ID and GITLAB_CLIENT_SECRET env
variables to be set.

The integration tests will deploy the locally defined bundle.yaml on microk8s,
configures the necessary gitlab-client-id and gitlab-client-secret config options
for gitlab-integrator. The integration tests will expect that the deployed units
become active, that FINOS Legend Engine, SDLC, and Studio are reachable through
their IPs and through the default legend-host external-hostname configured. The
Legend applications should be reachable through a newly configured external-hostname.

(cherry picked from commit 9bab2c4f9fb9f83504e52c3e6797cc75d3310fb0)

integration: Removes GITLAB_CLIENT_ID and GITLAB_CLIENT_SECRET env vars

The integration tests only test that the Legend applications are reachable,
which doesn't require valid GITLAB_CLIENT_ID and GITLAB_CLIENT_SECRET configurations.

This is needed because we won't be able to get the GITLAB_CLIENT_ID and
GITLAB_CLIENT_SECRET repository secrets on pull_request (they are fetched from the
repository that makes the pull request), which means that it would fail.

(cherry picked from commit 51a836914c5c2fbf1b7859d800ddd0a8d0b3f643)

claudiubelu commented:
You're missing the ``tox.ini`` file.

Typically, when code is moved from one branch to another, ``git cherry-pick -x commit_id`` should be used instead. This would have prevented the missing ``tox.ini`` file as well. In this scenario, since it would mean cherry-picking 2 commits (https://github.com/finos/legend-juju-bundle/commit/9bab2c4f9fb9f83504e52c3e6797cc75d3310fb0, https://github.com/finos/legend-juju-bundle/commit/51a836914c5c2fbf1b7859d800ddd0a8d0b3f643), I recommend squashing them first, and then cherry-pick your commit on top. It will also make it easier to review.

Commit:
Run tests periodically and send email on failure

The integration tests have been cherry-picked (and squashed) from the
`edge` branch.

smiclea commented:
@claudiubelu The PR has been updated as per your instructions.

fabi200123 commented on .github/workflows/ci.yaml:
```diff
@@ -0,0 +1,58 @@
+      - name: Setup operator environment
+        # TODO: change this to charmed-kubernetes/actions-operator@main once
+        # the following issue is addressed:
+        # https://github.com/charmed-kubernetes/actions-operator/issues/32
```
We should use `charmed-kubernetes/actions-operator@main`, since the issue mentioned there was solved in [here](https://github.com/charmed-kubernetes/actions-operator/pull/46). This will fix the problem with the `setup operator environment`

claudiubelu replied:
Thanks. This was solved on the ``edge`` branch before [1], so it's worth cherry-picking here as well.

[1] https://github.com/finos/legend-juju-bundle/pull/33

claudiubelu commented:
This will require a few other cherry-picks in order to pass, will be added shortly.
